### PR TITLE
Update script item style reccommendations

### LINKED ISSUE #149: Update script item indentation advice.
We originally recommended indenting the internal lines of multi-line script strings to the left margin, because they are verbatim bash job script inclusions and not part of the surrounding config file syntax:
```
[runtime]
   [[cheese]]
       script = """
echo HELLO
echo GOODBYE
"""
```

However, automatic dedent processing now means it is safe to do this  (even for heredocs - athough you have to be careful to get the indentation exactly right for the heredoc end marker):
```
[runtime]
   [[cheese]]
       script = """
           echo HELLO
           echo GOODBYE
       """
```
... which is generally considered to look better (even though it is technically wrong :stuck_out_tongue_winking_eye: )

Given that, we should:
- [ ] update indentation advice in the Suite Design Guide
- [ ] explain why it is OK to indent like this - for "the thinking user" :brain:

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #149

- [x] update indentation advice in the Suite Design Guide
- [x] explain why it is OK to indent like this - for "the thinking user" 🧠

### CODE REVIEW

File: src/workflow-design-guide/style-guide.rst [hunk 1]
```diff
@@ -92,45 +92,49 @@ to, and trailing comments should be preceded by two spaces, as shown above.
...
+
+.. code-block:: cylc
+
+   [runtime]
+       [[foo]]
+           script = echo "Hello, Mr. Thompson"
+
+If using a triple-quoted string, any common leading whitespace is trimmed
+using the logic of
+`Python's textwrap.dedent <https://docs.python.org/2/library/textwrap.html#textwrap.dedent>`_.
```
Comment (oliver-sanders) [inline src/workflow-design-guide/style-guide.rst:105]: We have hooked up the Python docs via intersphinx so you can use something like:

```rst
:py:func:`textwrap.dedent`
```

To reference `dedent` in the Python 3 docs.

File: src/workflow-design-guide/style-guide.rst [hunk 2]
```diff
@@ -143,8 +147,7 @@ have many levels of indentations.
                _EOF_
```
Comment (oliver-sanders) [inline src/workflow-design-guide/style-guide.rst:147]: Is this EOF aligned with the `cat`?

**Review by oliver-sanders — COMMENTED** [review]
Looks good.

File: src/workflow-design-guide/style-guide.rst (hunk 2 above)
Comment (MetRonnie) [reply src/workflow-design-guide/style-guide.rst:147]: yes

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**